core/drivers/clk/sub.mk

```diff
@@ -1,3 +1,5 @@
 srcs-y += clk.c
 srcs-$(CFG_DRIVERS_CLK_DT) += clk_dt.c
 srcs-$(CFG_DRIVERS_CLK_FIXED) += fixed_clk.c
+
+subdirs-y += sam
```

Comment:
i think it should better be here:
`subdirs-$(CFG_DRIVERS_SAM_CLK) += sam`

And in _clk/sam/sub.mk_ use a specific `CFG_SAMA5D2_CLK` config switch for _sama5d2_clk.c_ (or unconditionally embedded it since sama5d2 is the only existing flavor/variant of plat-sam)

Reply:
Ok